### PR TITLE
Wall index

### PR DESCRIPTION
**Pull request by andreorvalho**
This will provide all code necessary to show a wall and all its posts.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 Rails.application.routes.draw do
+  get 'wall/index'
```
Comment (andreorvalho) [inline config/routes.rb]: This is not necessary resources walls already does this

File: app/controllers/wall_controller.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class WallController < ApplicationController
+  def show
+    post = Post.find(params[:id])
```
Comment (andreorvalho) [inline app/controllers/wall_controller.rb]: What is that you are trying to achieve with this line? What is your goal?

File: app/models/wall.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
 class Wall < ActiveRecord::Base
   # Associations
-  has_many :posts, dependent: :destroy
+  has_many :posts, dependent: :destroy, order: 'dateg desc'
```
Comment (andreorvalho) [inline app/models/wall.rb]: What is dateg?

File: app/views/wall/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<h1>Wall#index</h1>
+<p>Find me in app/views/wall/index.html.erb</p>
+<% post.walls.each do |o| %>
+          <td><%= o.dateg %></td>
```
Comment (andreorvalho) [inline app/views/wall/index.html.erb]: indentation is wrong.
You are also missing an end here. 
it should be:

```erb
<% post.walls.each do |o| %>
<% end %>
```

File: db/schema.rb [hunk 5]
```diff
@@ -32,6 +32,14 @@
     t.datetime "updated_at"
   end
 
+  create_table "users", force: true do |t|
+    t.string   "first_name"
+    t.string   "last_name"
+    t.string   "email"
+    t.datetime "created_at"
+    t.datetime "updated_at"
+  end
```
Comment (andreorvalho) [inline db/schema.rb]: why and how did this get here?

File: spec/controllers/wall_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe WallController, :type => :controller do
+
+  describe "GET 'index'" do
+    it "returns http success" do
+      get 'index'
+      expect(response).to be_success
+    end
+  end
```
Comment (andreorvalho) [inline spec/controllers/wall_controller_spec.rb]: this tests the index action that you have not yet defined on the wall controller!

**Comment by andreorvalho** [comment]
> It has a syntax error in the "http://localhost:3000/wall/index"
<%= o.dateg %>
When I erased get 'wall/index' the result was an error

This sentence does not really make sense to me. But if you are trying to define the show action why are you accessing an url to the index?

File: db/migrate/20141012225644_create_posts.rb [hunk 7]
```diff
@@ -2,8 +2,8 @@ class CreatePosts < ActiveRecord::Migration
   def change
     create_table :posts do |t|
       t.text :content
-
       t.timestamps
+      add_index :posts, [:wall_id, :created_at]
     end
   end
 end
```
Comment (jonybravo) [inline db/migrate/20141012225644_create_posts.rb]: turn back...

File: app/models/post.rb [hunk 8]
```diff
@@ -4,4 +4,7 @@ class Post < ActiveRecord::Base
 
   # Validations
   validates :content, :wall, presence: true
+
+  # Chronologi
+  default_scope -> { order(created_at: :desc) }
 end
```
Comment (jonybravo) [inline app/models/post.rb]: goingo to erase and create a post controller with a "def index order"

File: app/models/wall.rb [hunk 9]
```diff
@@ -1,4 +1,4 @@
 class Wall < ActiveRecord::Base
   # Associations
-  has_many :posts, dependent: :destroy
+  has_many :posts, -> { order('created_at DESC') }, dependent: :destroy
```
Comment (andreorvalho) [inline app/models/wall.rb]: so this is still providing a default scope!

The problem is the following:
- Everytime I do wall.posts, those posts will always come ordered by created at...
- I want that everytime I do wall.posts that it does not come with any order.

this here is what I told you it was a scope: http://guides.rubyonrails.org/active_record_querying.html#scopes

This makes it possible for us to have something we can re-use on the controller when we need, it works like a method that you apply to a collection of object like the one wall.posts returns. I will commit the changes with the correct code

**Comment by andreorvalho** [comment]
@jonybravo I have squashed the commits and added corrections from my last comments.

I have also added the correct tests! please have a look! make all the questions when you feel confortable to go on I will merge the PR

**Pull request merged**